tempesta_fw/cache.c

```diff
@@ -252,10 +252,17 @@ tfw_cache_employ_req(TfwHttpReq *req)
 }
 
 static bool
-tfw_cache_employ_resp(TfwHttpReq *req)
+tfw_cache_employ_resp(TfwHttpResp *resp, TfwHttpReq *req)
 {
 	if (req->cache_ctl.flags & TFW_HTTP_CC_CFG_CACHE_BYPASS)
 		return false;
+	if (!TFW_STR_EMPTY(&req->h_tbl->tbl[TFW_HTTP_HDR_AUTHORIZATION]) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_PUBLIC) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_S_MAXAGE) &&
+	    !(resp->cache_ctl.flags & TFW_HTTP_CC_MUST_REV))
```

Comment:
RFC 2616, chapter 14.8 (see issue #3 ):
"If the response includes "s-maxage=0", the proxy MUST always revalidate it before re-using it."
I believe that this shows that the cache is possible, but must be revalidation